### PR TITLE
drivers: clock_control of the stm32l0x or stm32l1x devices with MSI

### PR DESCRIPTION
**Pull request by FRASTM**
Fix register bit field when clock source is MSI
on the stm32L0x or stm32L1x mcus
Use RCC_CR_MSIRGSEL bit field instead of soc is not stm32wbx serie

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

File: drivers/clock_control/clock_stm32_ll_common.c [hunk 1]
```diff
@@ -39,6 +39,9 @@
 #if defined(CONFIG_SOC_SERIES_STM32WBX)
 #define RCC_CALC_MSI_RUN_FREQ(range) __LL_RCC_CALC_MSI_FREQ( \
 						range << RCC_CR_MSIRANGE_Pos)
+#elif defined(CONFIG_SOC_SERIES_STM32L0X) || defined(CONFIG_SOC_SERIES_STM32L1X)
+#define RCC_CALC_MSI_RUN_FREQ(range) __LL_RCC_CALC_MSI_FREQ( \
+						range << RCC_ICSCR_MSIRANGE_Pos)
 #else
 #define RCC_CALC_MSI_RUN_FREQ(range) __LL_RCC_CALC_MSI_FREQ( \
 			LL_RCC_MSIRANGESEL_RUN, range << RCC_CR_MSIRANGE_Pos)
```
Comment (erwango) [inline drivers/clock_control/clock_stm32_ll_common.c:47]: Can this be simplified using `LL_RCC_MSI_GetRange` ?
```
#define RCC_CALC_MSI_RUN_FREQ(range) __LL_RCC_CALC_MSI_FREQ(LL_RCC_MSI_GetRange())
```

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/clock_control/clock_stm32_ll_common.c (hunk 1 above)
Comment (FRASTM) [reply drivers/clock_control/clock_stm32_ll_common.c:47]: though, the difference in shift operation appears clearly from the RCC register bit name, as it is written in the line above and the line below

Comment (erwango) [reply drivers/clock_control/clock_stm32_ll_common.c:47]: I mean: we can replace these 3 alternate definitions by a single one:
```
#define RCC_CALC_MSI_RUN_FREQ(range) __LL_RCC_CALC_MSI_FREQ(LL_RCC_MSI_GetRange())
```

Comment (FRASTM) [reply drivers/clock_control/clock_stm32_ll_common.c]: ok,

**Comment by FRASTM** [comment]
redefine  RCC_CALC_MSI_RUN_FREQ macro depending on the __LL_RCC_CALC_MSI_FREQ macro

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/clock_control/clock_stm32_ll_common.c [hunk 2]
```diff
@@ -36,12 +36,13 @@
...
-#define RCC_CALC_MSI_RUN_FREQ(range) __LL_RCC_CALC_MSI_FREQ( \
-						range << RCC_CR_MSIRANGE_Pos)
+#if !defined(LL_RCC_MSIRANGESEL_RUN)
+/* CONFIG_SOC_SERIES_STM32WBX or CONFIG_SOC_SERIES_STM32L0X or CONFIG_SOC_SERIES_STM32L0X */
+#define RCC_CALC_MSI_RUN_FREQ(range) __LL_RCC_CALC_MSI_FREQ(LL_RCC_MSI_GetRange())
 #else
+/* mainly CONFIG_SOC_SERIES_STM32WLX or CONFIG_SOC_SERIES_STM32L4X or CONFIG_SOC_SERIES_STM32L5X */
 #define RCC_CALC_MSI_RUN_FREQ(range) __LL_RCC_CALC_MSI_FREQ( \
-			LL_RCC_MSIRANGESEL_RUN, range << RCC_CR_MSIRANGE_Pos)
+			LL_RCC_MSIRANGESEL_RUN, LL_RCC_MSI_GetRange())
```
[the remark this answers is not in the document]
Comment (FRASTM) [reply drivers/clock_control/clock_stm32_ll_common.c]: removing arg

**Review by erwango — APPROVED** [review]

**Review by ABOSTM — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**